### PR TITLE
add the endpoint(6): GET /api/articles/:article_id/comments

### PR DESCRIPTION
**Pull request by Jared-Ha**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -143,3 +143,62 @@ describe("GET /api/articles", () => {
...
+			.expect(200)
+			.then(({ body: { comments } }) => {
+				comments.forEach((comment) => {
+					expect(comment).toEqual({
+						comment_id: expect.any(Number),
+						votes: expect.any(Number),
+						created_at: expect.any(String),
+						author: expect.any(String),
+						body: expect.any(String),
+						article_id: expect.any(Number),
```
Comment (MRombi) [inline __tests__/app.test.js:171]: This should be a specifically article id, not any number :)

File: __tests__/app.test.js [hunk 2]
```diff
@@ -143,3 +143,62 @@ describe("GET /api/articles", () => {
...
+
+describe("GET /api/articles/:article_id/comments", () => {
+	it("GET 200: responds with all comments that correspond to the given article ID'", () => {
+		return request(app)
+			.get("/api/articles/3/comments")
+			.expect(200)
+			.then(({ body: { comments } }) => {
+				expect(comments.length).toBe(2);
+				comments.forEach((comment) => {
+					expect(comment.article_id).toBe(3);
```
Comment (MRombi) [inline __tests__/app.test.js:155]: We do not need this test if we change the test below :)

File: __tests__/app.test.js [hunk 3]
```diff
@@ -143,3 +143,62 @@ describe("GET /api/articles", () => {
...
+	});
+	it("GET 200: repsonds with most recent comments first (i.e. descending date order)", () => {
+		return request(app)
+			.get("/api/articles/3/comments")
+			.expect(200)
+			.then(({ body: { comments } }) => {
+				for (let i = 1; i < comments.length; i++) {
+					expect(Date.parse(comments[i - 1].created_at)).toBeGreaterThanOrEqual(
+						Date.parse(comments[i].created_at)
+					);
```
Comment (MRombi) [inline __tests__/app.test.js:184]: We could use jest sorted instead :)

File: __tests__/app.test.js [hunk 4]
```diff
@@ -143,3 +143,62 @@ describe("GET /api/articles", () => {
...
+				}
+			});
```
Comment (MRombi) [inline __tests__/app.test.js:186]: We should add a test for when the article has no comments associated with it, like with article id 2 :)

**Review by MRombi — COMMENTED** [review]
Have a look at my feedback, well done overall 😄

File: __tests__/app.test.js (hunk 3 above)
Comment (Jared-Ha) [reply __tests__/app.test.js:184]: expect(comments).toBeSorted("created_at", { descending: true });

File: __tests__/app.test.js (hunk 4 above)
Comment (Jared-Ha) [reply __tests__/app.test.js:186]: tested and passing :) (used a util function)

	it("GET 200: responds with an empty array when an existing article ID is used, but there are no comments associated with that article yet", () => {
		return request(app)
			.get("/api/articles/2/comments")
			.expect(200)
			.then(({ body: { comments } }) => {
				expect(comments).toEqual([]);
			});
	});

**Pull request merged**